### ISSUE TITLE
Record is an ETAS record but is not showing the Hathitrust ETAS link (another one)

### ISSUE THREAD

**Issue by mkutch** [psu-libraries/psulib_blacklight#606]
Catkey `3148764` doesn't show ETAS link in catalog but does in the CAT. I checked Hathitrust and it has temporary access through ETAS.

**Comment by cdmo** [comment]
Our July overlap report that we get from HT says `61123117` maps to catkey `3148764`. The oclc number for this record is not in the hathi full report for July, so we throw the record out. `61123117` (the OCLC identifier for this record in our catalog) is not present in `hathi_full_20200701.txt`.

The record in HT’s catalog has an OCLC number of `61440858`, so there is a mismatch there.

What I don’t understand is how The CAT is getting something back from the API for this record, because The CAT is also using `61123117` as the OCLC. The call that The CAT performs for this record is:

`GET | https://catalog.hathitrust.org/api/volumes/brief/oclc/61123117.json?callback=jsonp1596120906537`

That should not resolve for this record, see the MARC: https://catalog.hathitrust.org/Record/005114789.marc.

Maybe we should contact Hathi? I can only guess that maybe this record was updated recently and instead of trashing the no-longer-correct response for the "old" OCLC number, HT just left it up (as a courtesy?). 

https://catalog.hathitrust.org/api/volumes/brief/oclc/61440858.json does resolve to the same data so that might be why?

But yeah, maybe lets ask Hathi whats up

**Comment by cdmo** [comment]
I bet the record changed OCLC number sometime after the July full and the time our overlap was generated for us.

**Comment by mkutch** [comment]
The json returned shows 4 oclc numbers associated with this record and one of them is `61123117`, so I think that must be how the CAT is getting a record back. Correct me if I'm wrong.

?({
  "records": {
    "005114789": {
      "recordURL": "https:\/\/catalog.hathitrust.org\/Record\/005114789",
      "titles": [
        "Strength training anatomy"
      ],
      "isbns": [
        "9780736063685",
        "9780736063685",
        "9780736063685"
      ],
      "issns": [
        
      ],
      "oclcs": [
        "61440858",
        "61123117",
        "224463427",
        "797140334"
      ],
      "lccns": [
        "2005020894"
      ],
      "publishDates": [
        "2005"
      ]
    }
  },
  "items": [
    {
      "orig": "University of Michigan",
      "fromRecord": "005114789",
      "htid": "mdp.39015062485548",
      "itemURL": "https:\/\/hdl.handle.net\/2027\/mdp.39015062485548",
      "rightsCode": "ic",
      "lastUpdate": "20170629",
      "enumcron": false,
      "usRightsString": "Limited (search-only)"
    }
  ]
})

**Comment by cdmo** [comment]
Yeah looks right. Just weird that there is only one oclc identifier on the record in Hathi. Yet it will apparently respond affirmatively for these other oclc numbers. Like I said earlier it looks like they keep around previous oclc numbers for the BibAPI responses. I haven't checked all the July update files yet. Will do in a bit

**Comment by cdmo** [comment]
Not in any of the update files. We just don't match on OCLC here. Do you know why the OCLC number we have for this record is different than the record that Hathi has? 

Basically like the OCLC The CAT queries on is an alias for the Hathi record, but, it does not show up in the marc records anywhere in the Hathi record for this item.

**Comment by jeffedmunds** [comment]
Having read through this thread several times, my best guess is that the HT BibAPI used for The CAT is running off different (older? or more current?) data than the data being queried by Blackcat (as reflected in the most current hathi full file). Either that or the BibAPI is functioning differently than whatever has been implemented in Blackcat.

To respond to @cdmo 's question above: the reason the OCLC number we have for this record is different from the OCLC number in the HT record is because OCLC is continuously merging records that its algorithm identifies as duplicates. In this case, there were (at least) 4 records that OCLC merged (whence the 4 OCLC numbers associated with this title); "old"  OCLC numbers are written to the 019 field for legacy purposes. Whence the challenges of keeping multiple silos "in sync" with respect to "unique" record IDs.

Is it possible that the BibAPI is not matching on the OCLC number on this case but on ISBN? Does the BibAPI test for multiple values in multiple fields (ISBN, OCLC number, barcode)? If yes, does Blackcat function in the same way, or does it ONLY look at OCLC numbers? This might provide clues to the answer in this case.

I did confirm that the OCLC record was last updated well over a year ago, which implies the collapse of the 4 records into 1 did not occur recently. So, in theory, if the BibAPI in The CAT and the ETAS implementation in Blackcat are both configured to operate in the same way (querying multiple data points when one is not present or returns multiple values) and if the data being queried on HT's end is the same, then The CAT and Blackcat should be giving identical results. If you want me to contact HT about this specific case, let me know.

**Comment by ruthtillman** [comment]
For some of the issues identified, are they in the ETAS matches at all with a our catkey? I'm wondering if (not emergency but soonish) we might need to revise the data to get the catkey vs OCLC, if HT is providing it. But I can't recall the various data structures offhand and whether they would support this. It would also require checking a handful of known problems so that we could determine the catkey was actually in them.

**Comment by cdmo** [comment]
The PSU catkey (`local_id`) is in the overlap report which we receive monthly along with the OCLC number. The only way we can crosswalk the overlap report over the hathi full extract (which allows us to retrieve the important information of `htid` and `ht_bib_key`) is to use the OCLC number because the hath full extract does not have the PSU catkey. Basically, OCLC is the sole way to "lookup" those values from the full. 

Now, if there was some other report from HT along the lines of "Alternate OCLCs and their identifiers" that would be a workable solution.

I did submit a ticket already to HT (did so last night). They've written back to say they are routing the issue to the appropriate party.

Maryam and I are still verifying the exact number of "misses" and will report back

**Comment by ruthtillman** [comment]
Thanks, I half-recalled that there was some reason that we might not be using it, but then I blanked on why. I agree that the additional report matching those keys would be helpful.

**Comment by cdmo** [comment]
To record that we just spoke via Zoom, now that we have figured which items are misses we will fix in the ingesting process, which will happen early next week.
